edi/models/edi_exchange_record.py

```diff
@@ -369,10 +372,16 @@ def check_access_rule(self, operation):
         if self.env.is_superuser():
             return
         for exchange_record in self:
-            if not exchange_record.record:
+            sudo_record = exchange_record.sudo()
```

Comment:
This is the same change applied on 12.0

Reply:
wouldn't be better do it before? `for exchange_record in self.sudo()`?

Reply:
Sure